src/sets.jl

```diff
@@ -807,6 +807,127 @@ end
 
 dimension(set::Complements) = 2 * set.dimension
 
+"""
+    AllDifferent(dimension::Int)
+
+The set corresponding to an all-different constraint.
+
+All expressions of a vector-valued function are enforced to take distinct
+values in the solution: for all pairs of expressions, their values must
+differ.
+
+This constraint is sometimes called `distinct`.
+"""
+struct AllDifferent <: AbstractVectorSet
+    dimension::Int
+end
+
+"""
+    Domain{T <: Real}(values::Set{T})
+
+The set corresponding to an enumeration of constant values.
+
+The value of a scalar function is enforced to take a value from this set of
+values.
+
+This constraint is sometimes called `in`.
+"""
+struct Domain{T <: Real} <: AbstractScalarSet
+    values::Set{T}
+end
+
+function Base.copy(set::Domain{T <: Real})
+    return Domain(copy(set.values))
+end
+
+"""
+    Membership(values::Set{AbstractScalarFunction})
+
+The set corresponding to an enumeration of values, given by expressions.
+
+The value of a scalar function is enforced to take a value from this set of
+values. These values are given by expressions potentially depending on other
+variables for the problem.
+
+This constraint is sometimes called `in_set`.
+"""
+struct Membership <: AbstractScalarSet
+    values::Set{AbstractScalarFunction}
+end
+
+function Base.copy(set::Membership)
+    return Domain(copy(set.values))
+end
+
+"""
+    DifferentFrom(value::AbstractScalarFunction)
+
+The set excluding the single point ``x \\in \\mathbb{R}`` where ``x`` is given by `value`.
+All other values in the domain of ``x`` are allowed. `value` is a generic scalar function.
+"""
+struct DifferentFrom <: AbstractScalarSet
+    value::AbstractScalarFunction
+end
+
+function Base.copy(set::DifferentFrom)
+    return DifferentFrom(copy(set.value))
+end
+
+"""
+    Count{T <: Real}(values::Set{AbstractScalarFunction}, value::T)
+
+``\\{(y, x) \\in \\mathbb{N} \\times \\mathbb{R}^n : y = |\\{i | x_i = value\\}|\\}``
+"""
+struct Count{T <: Real} <: AbstractVectorSet
+    value::T
+    dimension::Int
+end
+
+dimension(set::Count{T}) where T = set.dimension + 1
+
+function Base.copy(set::Count)
+    return Count(copy(set.value), value)
+end
+
+"""
+    StrictlyGreaterThan{T <: Real}(lower::T)
+
+The set ``(lower,\\infty) \\subseteq \\mathbb{R}``. It is mostly useful if `lower` is an integer.
+"""
+struct StrictlyGreaterThan{T <: Real} <: AbstractScalarSet
+    lower::T
+end
```

Comment:
Another option is 

```Julia
struct Strictly{S <: Union{LessThan, GreaterThan}} <: AbstractScalarSet
    set::S
end
```
So `Strictly(LessThan(1))`

Make it easier if we want to add `Strictly{Nonnegatives}`, `Strictly{Interval{Int}}`, etc.